Assets/Mirror/Runtime/NetworkConnection.cs

```diff
@@ -375,17 +393,17 @@ public virtual void TransportReceive(ArraySegment<byte> buffer)
         /// This virtual function allows custom network connection classes to process data send by the application before it goes to the network transport layer.
         /// </summary>
         /// <param name="channelId">Channel to send data on.</param>
-        /// <param name="bytes">Data to send.</param>
+        /// <param name="segment">Data to send. Will be recycled after returning, so use it directly.</param>
         /// <returns></returns>
-        public virtual bool TransportSend(int channelId, byte[] bytes)
+        public virtual bool TransportSend(int channelId, ArraySegment<byte> segment)
```

Comment:
why the heck is this a public method? 
Technically this is a breaking change because it is public,  but noone should be using this method.

add an [obsolete] method with old signature

Reply:
TransportReceive is public too...should both be private?

Reply:
they should at least be internal,   users should not be able to send random byte[]

Reply:
let's discuss public/internal in a separate pr. added obsolete though.